### PR TITLE
Migrate the debugger from python2 to python3

### PR DESCRIPTION
**Pull request by gergocs**
Signed-off-by: Gergo Csizi gergocs@inf.u-szeged.hu

### CODE REVIEW

**Comment by gergocs** [comment]
Patch is ready for review

File: tools/debugger/debugger_core.py [hunk 1]
```diff
@@ -930,7 +918,7 @@ def _parse_source(self, data):
             logging.debug("Pending breakpoints available")
             bp_list = self.pending_breakpoint_list
 
-            for breakpoint_index, breakpoint in bp_list.items():
+            for breakpoint_index, breakpoint in list(bp_list.items()):
```
Comment (zherczeg) [inline tools/debugger/debugger_core.py:921]: with a while loop this could be optimized

**Review by zherczeg — CHANGES_REQUESTED** [review]
Nice patch only a minor change

File: tools/debugger/debugger_core.py (hunk 1 above)
Comment (gergocs) [reply tools/debugger/debugger_core.py:921]: It's not possible

**Review by zherczeg — COMMENTED** [review]
LGTM (informal)

**Review by zherczeg — APPROVED** [review]

**Review by clover2123 — APPROVED** [review]

**Pull request merged**